src/io/flutter/view/InspectorPanel.java

```diff
@@ -89,15 +91,29 @@ public InspectorPanel(FlutterView flutterView,
     this.flutterView = flutterView;
     this.isApplicable = isApplicable;
 
-    myRootsTree = new MyTree(new DefaultMutableTreeNode(null));
+    myRootsTree = new TreeDataProvider(new DefaultMutableTreeNode(null), treeType.displayName);
     myRootsTree.addTreeExpansionListener(new MyTreeExpansionListener());
     myPropertiesPanel = new PropertiesPanel();
 
     initTree(myRootsTree);
     myRootsTree.getSelectionModel().addTreeSelectionListener(e -> selectionChanged());
-    // TODO(devoncarew): We should remember and restore the last splitter position.
+
     final Splitter treeSplitter = new Splitter(true);
-    treeSplitter.setProportion(0.8f);
+    treeSplitter.setProportion(flutterView.getState().getSplitterProportion());
+    flutterView.getState().addListener(e -> {
+      final float newProportion = flutterView.getState().getSplitterProportion();
+      if (treeSplitter.getProportion() != newProportion) {
+        treeSplitter.setProportion(newProportion);
+      }
+    });
+    //noinspection Convert2Lambda
+    treeSplitter.addPropertyChangeListener("proportion", new PropertyChangeListener() {
```

Comment:
nit: is there a reason this isn't a lambda?

Reply:
The line was just a bit too long and harder to read.